searchable_test.go

```diff
@@ -0,0 +1,276 @@
+// Copyright 2020-2022 CYBERCRYPT
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package encryptonize
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/cyber-crypt-com/encryptonize-lib/crypto"
+)
+
+func TestNewSearchable(t *testing.T) {
+	searchable := NewSearchable()
+	if len(searchable.mapping) != 0 {
+		t.Fatal("Index non-empty at initialization.")
+	}
+}
+
+func TestAdd(t *testing.T) {
+	searchable := NewSearchable()
+
+	rand := &crypto.NativeRandom{}
+	masterKey, err := rand.GetBytes(32)
+	if err != nil {
+		t.Fatalf("Random failed: %v", err)
+	}
```

Comment:
It's up to you, but I'd say it's okay to skip the error handling (in tests!) for things you "know" cannot fail. 
```suggestion
	masterKey, _ := rand.GetBytes(32)
```